docs/evaluate/development-production-features/workflow-message-passing.mdx

```diff
@@ -15,34 +16,41 @@ keywords:
   - business logic activities
   - temporal workers
   - temporal SDK tutorial
-  - go sdk temporal guide
-  - java sdk temporal guide
-  - python sdk temporal guide
-  - typescript sdk temporal guide
   - learning temporal workflows
   - developing with temporal
   - temporal workflow execution
   - temporal activity management
   - worker process execution
   - signals
   - queries
+  - updates
 ---
 
 import { RelatedReadContainer, RelatedReadItem } from '@site/src/components/related-read/RelatedRead';
 
-Need a responsive Workflow?
-Temporal provides **Signals**, **Queries**, and **Updates** as a means to send data into and get data out of your running Workflow.
-Signals enable your Workflow to asynchronously react to events at runtime, Queries enable on-demand data retrieval from ongoing Workflows without disrupting their execution, and Updates enable you to change the state of a running Workflow.
+Need a responsive Workflow? Think about use cases like these:
 
-**Signals**: Signals allow you to send messages asynchronously to running Workflow Executions, enabling you to change their state and control their flow in real-time.
-This feature is particularly useful in scenarios where Workflows must react to external events or user interactions.
+- Your shipment-tracking Workflow needs to know when the item leaves the warehouse and is loaded into their truck. **Signal** your Workflow when the truck driver scans the barcode.
+- Folks in your company want to track the progress of their data migration Workflows. **Query** your running batch Workflow to get the data for the progress bar.
+- Your eCommerce shopping cart Workflow needs to know when a new item is added. **Update** it to add the item and receive back the current items to render on the next page.
 
-**Queries**: Queries provide a way to access and monitor the internal state of your Workflow Executions in real-time.
-This is essential for monitoring the progress of long-running Workflows and retrieving their results.
+Temporal provides Signals, Queries, and Updates to allow rich interactivity with your running Workflows.
 
-**Updates**: An Update is a synchronous, blocking call that could replace both a Signal and a Query.
+**Signals**: Signal to send messages asynchronously to a running Workflow, perhaps changing its state or controlling its flow in real-time.
```

Comment:
```suggestion
**Signals**: Signal to send messages asynchronously to a running Workflow, changing its state or controlling its flow in real-time.
```

Suggest deleting "perhaps" since if a signal does neither of those it is pointless.